omod/src/main/java/org/openmrs/web/controller/ConceptFormValidator.java

```diff
@@ -113,4 +119,78 @@ else if (backingObject.getNamesByLocale().get(locale).getConceptNameId() != null
 		}
 	}
 	
+	/**
+	 * Validates reference range fields
+	 * 
+	 * @param concept concept
+	 * @param errors errors
+	 * @since 1.17.0
+	 */
+	public void validateConceptReferenceRange(Concept concept, BindException errors) {
+		if (concept.isNumeric()) {
+			ConceptNumeric conceptNumeric = (ConceptNumeric) concept;
+			
+			List<ConceptReferenceRange> referenceRanges = new ConceptFormMapper().mapToWebReferenceRanges(conceptNumeric);
+			
+			if (referenceRanges == null || referenceRanges.isEmpty()) {
+				return;
+			}
+			
+			int index = 0;
+			for (ConceptReferenceRange referenceRange : referenceRanges) {
+				
+				if (referenceRange.getId() == null) {
+					if (referenceRange.getHiAbsolute() == null) {
+						setReferenceRangeErrors(errors, index, "hiAbsolute",
+						    "Concept.referenceRanges.error.high.absolute.value.required",
+						    "Concept.referenceRanges.error.absolute.value.required");
+					} else {
+						if (referenceRange.getHiAbsolute() > conceptNumeric.getHiAbsolute()) {
```

Comment:
What happens when conceptNumeric.getHiAbsolute() is null?

Reply:
Fixed